### PR TITLE
files / fileUploadInput function

### PR DESCRIPTION
**Pull request by pvilalta**
https://platform-dev-input-file.bimdata.io

hello team,

This PR brings the replacement of FileUploadButton component by the fileUploadInput function

Paul

### CODE REVIEW

File: src/views/project-board/project-bcf/ProjectBcf.vue [hunk 1]
```diff
@@ -447,7 +453,10 @@ export default {
       open: openSettings
     } = useToggle();
 
-    const importBcfTopics = async files => {
+    const importBcfTopics = async event => {
+      const files = Array.from(event.target.files);
+      if (files.length === 0) return;
```
Comment (Amoki) [inline src/views/project-board/project-bcf/ProjectBcf.vue]: Is it possible to have 0 files from the file input?
On firefox, I can't validate the form without a file

Comment (NicolasRichel) [reply src/views/project-board/project-bcf/ProjectBcf.vue]: I agree, I don't think it is necessary to check files length.

File: src/components/specific/files/file-uploader/FileUploader.vue [hunk 2]
```diff
@@ -88,13 +93,16 @@ export default {
...
     const uploadFiles = event => {
+      const targetFiles = Array.from(event.target.files);
+      if (targetFiles.length === 0) return;
+
       let files = null;
-      if (event.dataTransfer) {
+      if (targetFiles.dataTransfer) {
         // Files from drag & drop
-        files = Array.from(event.dataTransfer.files);
+        files = Array.from(targetFiles.dataTransfer.files);
```
Comment (NicolasRichel) [inline src/components/specific/files/file-uploader/FileUploader.vue]: The drag & drop is broken here...
As Hugo pointed out, I suggest to remove the files length check above and just keep the `event` variable as before.
Let me know if I'm missing something.

File: src/components/specific/versioning/versioning-main/VersioningMain.vue [hunk 3]
```diff
@@ -117,18 +121,19 @@ export default {
...
-            await getAllDocVersions(newHeadVersion);
-            loading.value = null;
-          }
-        };
-        const uploader = projectFileUploader(props.project, handlers);
-        uploader.upload(fileToUpload[0], null, allDocVersions.value[0].id);
-      }
+    const addVersion = async event => {
+      const files = Array.from(event.target.files);
+      if (files.length === 0) return;
```
Comment (NicolasRichel) [inline src/components/specific/versioning/versioning-main/VersioningMain.vue]: Same here, this check is not necessary.

**Review by NicolasRichel — CHANGES_REQUESTED** [review]

**Review by NicolasRichel — APPROVED** [review]

**Pull request merged**